### ISSUE TITLE
Remote LES drops connection after local client submits transaction

### ISSUE THREAD

**Issue by veox** [ethereum/go-ethereum#15943]
#### System information

Local light client:

OS & Version: Ubuntu 16.04.3 LTS
Geth version: `v1.7.3-stable-4bb3c89d` (via Ubuntu PPA)

Remote light server:

OS & Version: Ubuntu 16.04.3 LTS
Geth version: `v1.8.0-unstable-02aeb3d` (self-built)
Go: 1.9.2

#### Expected behaviour

Remote LES peers remain connected after local light client broadcasts a transaction.

#### Actual behaviour

Remote LES servers with versions >= v1.7.3 routinely disconnect after my local LES client broadcasts a transaction through them.

Observed with remote servers of v1.7.3 and v1.8.0 (unstable). So far seems not to affect v1.7.2 remotes.

#### Steps to reproduce the behaviour

* Wait until local light client connects to a v1.7.3 or v1.8.0 remote (console: `admin.peers`).
* Turn on debugging (console: `debug.vmodule("p2p=5")`).
* Send a transaction (console: `eth.sendTransaction({from: ..., to: ..., value: ..., gasPrice: ...})`.

#### Log

In the client's log (via `journalctl`, since running `geth` using `systemd`):

````
...
Jan 22 21:00:14 egs geth[3452]: INFO [01-22|21:00:14] Submitted transaction                    fullhash=<snipped> recipient=<snipped>
Jan 22 21:00:14 egs geth[3452]: DEBUG[01-22|21:00:14] Removing p2p peer                        id=138ec659b5ec0202 conn=inbound duration=17m25.607s peers=1 req=true  err="disconnect requested"
Jan 22 21:00:14 egs geth[3452]: TRACE[01-22|21:00:14] New dial task                            task="staticdial 138ec659b5ec0202 :::30303"
Jan 22 21:00:14 egs geth[3452]: TRACE[01-22|21:00:14] Accepted connection                      addr=[::1]:34494
Jan 22 21:00:14 egs geth[3452]: TRACE[01-22|21:00:14] Failed RLPx handshake                    addr=[::1]:34494                       conn=inbound err="ecies: invalid message"
Jan 22 21:00:14 egs geth[3452]: TRACE[01-22|21:00:14] Failed RLPx handshake                    addr=[::1]:30303                       conn=staticdial err=EOF
Jan 22 21:00:14 egs geth[3452]: TRACE[01-22|21:00:14] Dial task done                           task="staticdial 138ec659b5ec0202 :::30303"
...
````

That is, right after the the transaction is submitted, the remote requests a disconnect. (Any other modules than `p2p` I should make more verbose?)

In this case, the `138ec659b5ec0202` peer also happens to be mine (its version info is at the top).

**Comment by veox** [comment]
Confirming issue fixed after PR #15974.

`Geth/v1.8.0-unstable-722bac84/linux-amd64/go1.9.3` peers - one commit after PR merge - don't drop the connection. (Now tested with same version for both client and server.)

**Comment by karalabe** [comment]
Thanks for the confirm @veox !
